gradle/publish.gradle

```diff
@@ -18,16 +19,31 @@ publishing {
     publications {
         bintray(MavenPublication) {
             from components.java
-            artifact(sourceJar) {
-                classifier 'sources'
-            }
-            artifact(javadocJar) {
-                classifier 'javadoc'
-            }
+            artifact(sourceJar)
+            artifact(javadocJar)
+            version project.version.replaceAll(/-.*/, '-SNAPSHOT')
         }
     }
 }
 
+
+artifactory {
+    publish {
+        contextUrl = 'https://oss.jfrog.org/artifactory'
+        repository {
+            repoKey = (project.version =~ /-/ ? 'oss-snapshot-local' : 'oss-release-local')
+            username = System.env.BINTRAY_USER
+            password = System.env.BINTRAY_KEY
+            maven = true
+        }
+        defaults {
+            publications (publishing.publications.bintray)
+            properties = ['git': project.version]
+        }
+    }
+}
+artifactoryPublish.dependsOn 'generatePomFileForJfrogPublication', 'build'
```

Comment:
`generatePomFileForBintrayPublication` ... it's actually some string patterning that the publish plugin uses to create the task names